plugins/modules/pnp_intent.py

```diff
@@ -793,16 +825,50 @@ class instance for further use.
                 params=planned_count_params,
             )
 
+            dev_details_response = self.dnac_apply['exec'](
+                family="device_onboarding_pnp",
+                function="get_device_by_id",
+                params={"id": self.have["device_id"]}
+            )
+
+            pnp_state = dev_details_response.get("deviceInfo").get("state")
+
             if not self.want["site_name"]:
                 self.result['response'] = self.have.get("device_found")
                 self.result['msg'] = "Device is already added"
```

Comment:
I could see more statements are in else part and then we immediately return self. Instead, can you return self here itself.  We can remove else.